Objects/codeobject.c

```diff
@@ -2268,3 +2268,76 @@ _PyStaticCode_InternStrings(PyCodeObject *co)
     }
     return 0;
 }
+
+PyCodeObject *
+_Py_MakeShimCode(const _PyShimCodeDef *codedef)
+{
+    PyObject *name = NULL;
+    PyObject *co_code = NULL;
+    PyObject *lines = NULL;
+    PyCodeObject *codeobj = NULL;
+    uint8_t *loc_table = NULL;
+
+    name = _PyUnicode_FromASCII(codedef->cname, strlen(codedef->cname));
+    if (name == NULL) {
+        goto cleanup;
+    }
+    co_code = PyBytes_FromStringAndSize(
+        (const char *)codedef->code, codedef->codelen);
+    if (co_code == NULL) {
+        goto cleanup;
+    }
+    int code_units = codedef->codelen/2;
+    int loc_entries = (code_units + 7)/8;
```

Comment:
Maybe add a comment like "The maximum number of code units encoded by a single line table entry is 8. This is equivalent to `ceil(code_units / 8)`."